core/store/src/trie/mod.rs

```diff
@@ -757,6 +760,97 @@ impl Trie {
     pub fn get_trie_nodes_count(&self) -> TrieNodesCount {
         self.storage.get_trie_nodes_count()
     }
+
+    /// Given a root and a key, it returns a MerkleProof for that key.
+    ///
+    ///  Whenever a proof is not found, an empty vector is returned.
+    ///
+    /// # Errors
+    ///
+    /// If the root is not found on the trie, a `StorageError` will be returned
+    pub fn get_proof(
+        &self,
+        root: &CryptoHash,
+        key: &[u8],
+    ) -> Result<Vec<TrieProofItem>, StorageError> {
+        let mut key = NibbleSlice::new(key);
+
+        let mut hash = *root;
```

Comment:
Given that we are make a copy of root and key here, I wonder if it makes sense to change the API of the function to take ownership of these values.  Then the copying will happen at the caller but maybe the caller does not need access to the values after calling this function we might save a copy.

Reply:
I was being consistent with
```rs
    pub fn get(&self, root: &CryptoHash, key: &[u8]) -> Result<Option<Vec<u8>>, StorageError> {

```
Have no preferences

Reply:
I suppose being consistent is preferred then.  If we decide to change, we should change both APIs at the same time.